runtime/commonMain/src/kotlinx/serialization/internal/SerialClassDescImpl.kt

```diff
@@ -50,18 +66,15 @@ open class SerialClassDescImpl @JvmOverloads constructor(
     }
 
     override fun getElementDescriptor(index: Int): SerialDescriptor {
-        // todo: cache
-        return generatedSerializer?.childSerializers()?.getOrNull(index)?.descriptor
-                ?: descriptors.getOrNull(index)
-                ?: throw MissingDescriptorException(index, this)
+        return generatedSerializer?.childSerializers()?.get(index)?.descriptor ?: descriptors[index]
```

Comment:
It's still `MissingDescriptorExc` if the index is valid according to `elementsCount`, but descriptor was not inserted by user (when this class was used in hand-written custom serializer)

Reply:
I'll provide a proper replacement for custom serializers in a separate PR a bit later.